ksql-common/src/main/java/io/confluent/ksql/function/types/MapType.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.types;
+
+import java.util.Objects;
+
+public final class MapType extends ObjectType {
+
+  private final ParamType value;
+
+  private MapType(ParamType value) {
+    this.value = value;
```

Comment:
Do we support Maps of Maps? This would allow it. Perhaps we should check value is of one of the permitted ParamTypes for Map values.

Reply:
I think we do support maps of maps - and if we don't it's not because of the UDF type system, it's something else (and that should be in charge of failing)